Adding splash screen

mohammedhemaid opened the pull request:
Adding a splash screen

mohammedhemaid committed:
Add Splash Activity

mohammedhemaid committed:
Edit Splash screen logo

Gear61 commented on app/build.gradle:
```diff
@@ -48,6 +48,7 @@ dependencies {
     implementation 'androidx.cardview:cardview:1.0.0'
     implementation 'androidx.recyclerview:recyclerview:1.1.0-rc01'
     implementation 'com.jakewharton:butterknife:10.1.0'
+    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
```
We don't need constraint layout for this. I try to avoid constraint layout since I find its API to be confusing.

Gear61 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -1,56 +1,50 @@
 <?xml version="1.0" encoding="utf-8"?>
-<manifest
```
Can you undo all your AndroidManifest changes minus the new splash activity? There's a lot of new labels on activities and stuff too; we need to keep this code review focused.

Gear61 commented on app/src/main/res/drawable/splash_screen.xml:
```diff
@@ -0,0 +1,12 @@
+    <item>
+        <bitmap
+            android:gravity="center"
+                android:src="@drawable/splash_logo"
```
nit: Make the spacing between src and gravity consistent. Looks like we just need to indent back a tab?

Gear61 replied:
Reminder to follow up on this.

Gear61 commented on app/app.iml:
```diff
@@ -94,6 +94,7 @@
     <orderEntry type="library" name="Gradle: androidx.arch.core:core-common:2.1.0@jar" level="project" />
     <orderEntry type="library" name="Gradle: com.jakewharton:butterknife-annotations:10.1.0@jar" level="project" />
     <orderEntry type="library" name="Gradle: androidx.annotation:annotation:1.1.0@jar" level="project" />
+    <orderEntry type="library" name="Gradle: androidx.constraintlayout:constraintlayout-solver:1.1.3@jar" level="project" />
```
Hopefully these disappear once we remove constraint layout.

Gear61 reviewed (changes requested):
This is a good start! It looks like there's some weird formatting stuff going on, so the main thing is to fix that. Also, feel free to increase the version code in the app level's build.gradle as well.

mohammedhemaid committed:
Adding Splash Screen

-Fixed formatting issue in manifest.
-Edited logo size.
-Removed constraint layout

mohammedhemaid committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	app/src/main/res/drawable-hdpi/splash_logo.png
#	app/src/main/res/drawable-mdpi/splash_logo.png
#	app/src/main/res/drawable-xhdpi/splash_logo.png
#	app/src/main/res/drawable-xxhdpi/splash_logo.png
#	app/src/main/res/drawable-xxxhdpi/splash_logo.png

mohammedhemaid commented:
Sorry for these mistakes I will work hard on my Git and Github to be familiar with it.

Gear61 commented:
@mohammedhemaid No need to apologize, the main point of code review is to make mistakes and learn from it!

Gear61 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -1,11 +1,11 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest
-    xmlns:android="http://schemas.android.com/apk/res/android"
+xmlns:android="http://schemas.android.com/apk/res/android"
```
nit: Move the spacing to where it was before.

Gear61 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -1,11 +1,11 @@
 
-    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
-    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
+    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
```
nit: Remove the extra spaces at the end for this line and the next one.

Gear61 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -15,42 +15,46 @@
         android:allowBackup="true"
         android:icon="@mipmap/ic_launcher"
         android:theme="@style/AppTheme">
```
nit: Add newline after this line.

Gear61 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -15,42 +15,46 @@
         android:allowBackup="true"
         android:icon="@mipmap/ic_launcher"
         android:theme="@style/AppTheme">
+        <activity android:name=".init.SplashActivity"
```
nit: Move android:name to the next line.

Gear61 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -15,42 +15,46 @@
+
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+
```
nit: Delete the newline here and the spaces after MAIN and LAUNCHER.

Gear61 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -15,42 +15,46 @@
-            android:configChanges="orientation|screenSize">
+            android:configChanges="orientation|screenSize"
+            android:label="@string/import_name_list"
+            android:windowSoftInputMode="adjustResize">
```
Ahhh, I realize now that all of these rearrangements are for alphabetizing. That makes sense; good change :)

Gear61 reviewed (changes requested):
Thanks for responding to the feedback! A couple minor formatting changes and we should be good.

mohammedhemaid committed:
Rearranged the code and removed the extra spaces

mohammedhemaid committed:
Rearranged the code and removed the extra spaces

mohammedhemaid committed:
Rearranged the code and removed the extra spaces

Gear61 reviewed (approved):
Looks good! Thanks for doing this!
